flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/sink/FileSink.java

```diff
@@ -141,13 +145,18 @@ private FileSink(BucketsBuilder<IN, ? extends BucketsBuilder<IN, ?>> bucketsBuil
     }
 
     @Override
-    public FileWriter<IN> createWriter(InitContext context) throws IOException {
+    public SinkWriter<IN> createWriter(InitContext context) throws IOException {
+        throw new UnsupportedOperationException("Not supported");
+    }
```

Comment:
Do we need to keep this implementation?

Reply:
This one is needed until we remove the method the deprecated method from the `Sink` interface.
We needed to keep this for backward compatibility, so if someone used the `Sink` as a functional interface then their code should keep working.